core/src/starknet_client/client.rs

```diff
@@ -159,8 +158,9 @@ impl StarknetClientConfig {
             return None;
         }
 
-        let transport =
-            HttpTransport::new(Url::parse(&self.starknet_json_rpc_addr.clone().unwrap()).ok()?);
+        let transport = HttpTransport::new(
+            ReqwestUrl::parse(&self.starknet_json_rpc_addr.clone().unwrap()).ok()?,
```

Comment:
nit: don't need to `clone` here, just do `self.starknet_json_rpc_addr.as_ref().unwrap()`